browser(webkit): rebase to 05/03/22 (r293746)

dpino opened the pull request:
Rebase `webkit/patches/bootstrap.diff` to [r293746](https://trac.webkit.org/changeset/293746/webkit). Changes:

  - [Rebase #1637 to r293746](https://github.com/dpino/WebKit/commit/4465be258e5a2e26a605851646a1405ffe58aded)
  - [Resolve conflicts](https://github.com/dpino/WebKit/commit/40bbbcda71d9c3c1749529dac15a039786e97b76)
  - [[WPE] Fix build after 250084@main](https://github.com/dpino/WebKit/commit/a8d00512549ecdc014b69258b7acc05ba2a25243)

yury-s commented on browser_patches/webkit/patches/bootstrap.diff:
```diff
@@ -1861,9 +1838,9 @@ index fc6fd38ecb392e9d5ea54d5702578562289434fb..6d4be8e5288a5a3307d7803246dcc16b
 +        m_timeZoneCache = std::unique_ptr<OpaqueICUTimeZone, OpaqueICUTimeZoneDeleter>(bitwise_cast<OpaqueICUTimeZone*>(timezone));
```
This if block should now be removed.

yury-s commented on browser_patches/webkit/patches/bootstrap.diff:
```diff
@@ -2241,26 +2218,11 @@ index 006b4e80c600158637c5d2f148975b758bd0d094..be517dde6a8877ab3525ea24bb76b94d
-index 735467108e3040bb1d23e9ba1a7ab8a2141a8f66..af5bde4f1c547142b2eaf9b17b59d4f30902c0bf 100644
+index 3e7644d1d232f8fc1a6b1572189dd366445a4e70..4234cab369a6d72fbc233c687a54b67221fa54c4 100644
 --- a/Source/WTF/wtf/DateMath.cpp
 +++ b/Source/WTF/wtf/DateMath.cpp
```
Now that timezone override logic is available upstream we revert all our changes to Source/WTF/wtf/DateMath.{h,cpp}. I can do this in a follow-up PR if you don't want to do it as part of the roll.

dpino committed:
browser(webkit): rebase to 05/03/22 (r293746)

dpino commented:
Addressed comments. Pushed 3 additional commits:
  - [Revert all downstream changes in DateMath.{cpp,h}](https://github.com/dpino/WebKit/commit/bdf450fbde6cce82251fa1bf0f2a6e60fb9ea710)
  - [Fix build error after removal of 'timeZoneForAutomation()'](https://github.com/dpino/WebKit/commit/4404a9022cc93d612559964f1e29f613310e9379)
  - [Fix build error after removal of 'setTimeZoneForAutomation'](https://github.com/dpino/WebKit/commit/1e0ee7e4f8dc4c9c5588cdb1e869b4d2a0bb6a4e)
